test/test_hyperframe.py

```diff
@@ -40,6 +40,25 @@ def test_repr(self, monkeypatch):
         monkeypatch.setattr(Frame, "serialize_body", lambda _: "A"*105)
         assert repr(f) == "Frame(Stream: 0; Flags: None): {}...".format("A"*100)
 
+
+class TestFlags(object):
+    def test_general_behaviour(self):
+        f = DataFrame(1)
+        assert not f.flags
+
+        f.flags.add("END_STREAM")
+        assert "END_STREAM" in f.flags
+        assert list(f.flags) == ["END_STREAM"]
+        assert len(f.flags) == 1
+
+        f.flags.discard("END_STREAM")
+        assert "END_STREAM" not in f.flags
+        f.flags.discard("END_STREAM")
+
+        with pytest.raises(ValueError):
+            f.flags.add("END_HEADERS")
+
```

Comment:
Can we split this into multiple tests? Ideally, if one of the tests for this structure fails I would know exactly what went wrong. It looks like we have a few tests here: test the basic adding logic, test the removal logic (probably still asserts length and conversion to list), and tests invalid insertions.